src/main/kotlin/minesweeper/domain/board/MineBoard.kt

```diff
@@ -1,35 +1,36 @@
 package minesweeper.domain.board
 
-import minesweeper.domain.cell.CellList
-import minesweeper.domain.cell.Empty
-import minesweeper.domain.cell.Mine
-import minesweeper.domain.common.NumberSet
-import minesweeper.domain.common.Position
+import minesweeper.domain.board.strategy.MineStrategy
+import minesweeper.domain.cell.Cells
+import minesweeper.domain.common.PositiveInt
+import minesweeper.domain.common.contains
+import minesweeper.domain.common.rangeTo
 
-class MineBoard(
-    val board: Board,
-    val mineIndices: NumberSet
+class MineBoard private constructor(
+    val width: PositiveInt,
+    val height: PositiveInt,
+    numberOfMines: PositiveInt,
+    mineStrategy: MineStrategy
```

Comment:
```suggestion
class MineBoard private constructor(
    val cells: Cells
) {
```

MineBoard가 cells만 가지고 있게하면 어떨까요?

cells만 가지게 된다면, 팩토리 메서드에서 cells를 만들어야할거에요.

```

companion object {
        fun of(
            width: Int,
            height: Int,
            numberOfMines: Int,
            mineStrategy: MineStrategy
        ): MineBoard {
            val cells = Cells.of(width, height, numberOfMines, mineStrategy)
            return MineBoard(cells)
        }
    }
```

Reply:
안녕하세요 형준님! 👋

코멘트 주신대로 개선해보았습니다~ 다만, 기존에 MineBoard init{} 블럭 내부에 있던 require 검증 로직들을 of() 함수 내부로 옮겼는데요. 이 부분 확인 한번만 부탁드립니다~! (width, height 을 임시 값으로만 받다보니, of 함수 내부에서만 검증이 가능하게 되었습니다..!)